cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -216,6 +216,95 @@ public List<LocationGroup> getLocationGroups(@Nullable String officeId,
         }
     }
 
+    /**
+     * Get all location groups for a given office and category,
+     * as well as a where clause to filter the shared_ref_location_id.
+     * @Param officeId The office id to use for the query.
+     * @Param locCategoryLike A regex to use to filter the location categories.  May be null.
+     * @Param sharedRefLocLike A where clause to filter the shared_loc_alias_id.  May be null.
+     * @return A list of all location groups for the given parameters.
+     */
+
+    public List<LocationGroup> getLocationGroups(String officeId, String locCategoryLike, String sharedRefLocLike) {
+        AV_LOC_GRP_ASSGN alga = AV_LOC_GRP_ASSGN.AV_LOC_GRP_ASSGN;
+        AV_LOC_CAT_GRP alcg = AV_LOC_CAT_GRP.AV_LOC_CAT_GRP;
+
+        final RecordMapper<Record, Pair<LocationGroup, AssignedLocation>> mapper = grpRecord -> {
+            LocationCategory category = buildLocationCategory(grpRecord);
+
+            LocationGroup group = buildLocationGroup(grpRecord, category);
+            AssignedLocation loc = buildAssignedLocation(grpRecord);
+
+            return new Pair<>(group, loc);
+        };
+
+        Map<LocationGroup, List<AssignedLocation>> map = new LinkedHashMap<>();
```

Comment:
This variable is declared and instantiated way before it is actually used

Reply:
Moved closer to use